woof-code/rootfs-petbuilds/firewallstatus/firewallstatus-0.7/firewallstatus.c

```diff
@@ -21,8 +22,11 @@ void quit(GtkWidget *w, gpointer dummy);
 void showme_window(GtkWidget *w, gpointer dummy);
 
 unsigned int interval = 2000; /*update interval in milliseconds */
+unsigned int long_interval = 600*1000;
+unsigned int max_tries = 30;
```

Comment:
After polling the current status 30 times (once every 2s), we start polling once every 10m. We assume that 1) rc.firewall will notify us if something has changed 2) it takes up to 1 minute for the firewall to start on boot and 3) users start and stop the firewall via rc.firewall.